include/RAJA/policy/atomic_builtin.hpp

```diff
@@ -37,389 +44,1368 @@ namespace RAJA
 struct builtin_atomic {
 };
 
-namespace detail
+
+namespace detail {
+
+
+#if defined(RAJA_COMPILER_MSVC) || (defined(_WIN32) && defined(__INTEL_COMPILER))
+
+
+/*!
+ * Atomic load
+ */
+RAJA_INLINE char builtin_atomicLoad(char *acc)
```

Comment:
Could we do something like a type traits based on size to avoid repeating so much code?

Reply:
For windows it's harder because the intrinsics are named differently for each size. I'll see what I can do, though.

Reply:
Would you be opposed to me using macros for some of this?

Reply:
Ya I think that would be fine, given how boiler-plate these functions all are.

Reply:
@rhornung67 What do you think?

Reply:
I've gotten rid of a ton of duplication just by adding some type traits and restructuring. I'm pretty satisfied with the new implementation. Let me know what you think.

Reply:
I think it looks good. No suggestions come to my mind.